http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/HttpVersionClientWithConfigKey.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import io.smallrye.mutiny.Uni;
+
+@RegisterRestClient(configKey = "http-client-with-config-key")
+@Path("/httpVersion")
+public interface HttpVersionClientWithConfigKey {
+
+    @GET
+    @Path("synchronous")
+    Response getClientHttpVersion(@QueryParam("name") String name);
+
+    @GET
+    @Path("asynchronous")
+    Uni<Response> getClientHttpVersionAsync(@QueryParam("name") String name);
```

Comment:
Hi @jedla97 , I noticed some code duplication in the method interfaces between HttpVersionClientWithConfigKey and HttpVersionClient. Perhaps it could be improved by applying inheritance to a single interface containing the methods getClientHttpVersion and getClientHttpVersionAsync, and then having the rest of the interfaces extend from this "BaseInterface" with those methods.

Reply:
@jcarranzan updated. I done it between the `ClientWithConfigKey` interfaces but not the rest. I agree that inherit it is better way. The `HttpVersionClientWithConfigKey` and `HttpsVersionClientWithConfigKey` inheriting from `HttpVersionClient` as they do same thing and are just specify name of `RestClient`. 

Personally don't want to introduce one new interface (base interface) and have 3 another interfaces inheriting from it. I see this base interface as something which are not needed + the inheritance from `HttpVersionClient` make sense for me.

Reply:
Thank you now it make sense to me also :+1: